pkg/s3/errors.go

```diff
@@ -0,0 +1,15 @@
+package s3
+
+import "github.com/bacalhau-project/bacalhau/pkg/models"
+
+const S3 = "S3"
+
+const (
+	S3BadRequest = "S3BadRequest"
+)
+
+func NewBadS3RequestError(message string) *models.BaseError {
+	return models.NewBaseError(message).
+		WithCode(S3BadRequest).
+		WithComponent(S3)
```

Comment:
using S3 as a component name can be a little ambiguous. Might be useful to understand if the failure happened with S3 storage source as it indicates an issue with job inputes, s3 publisher as it indiciates an issue with outputs, or S3 result signer as it indicates an issue with downloading results

Reply:
Fixed.